### PR TITLE
test: controller without reconcile

### PR DESCRIPTION
**Pull request by pkliczewski**
This PR provides over 70% test coverage to contoller code.

Signed-off-by: Piotr Kliczewski <piotr.kliczewski@gmail.com>

### CODE REVIEW

**Review by jakub-dzon — COMMENTED** [review]
Looks good!:+1:

What do you think about checking that status conditions are patched properly in all the places - maybe in a separate PR? 
I think it might be helpful to have the whole state machine under control.

**Comment by pkliczewski** [comment]
> Looks good!
> 
> What do you think about checking that status conditions are patched properly in all the places - maybe in a separate PR?
> I think it might be helpful to have the whole state machine under control.

I like the idea. My intention for this PR was to have working mocks and cover as much code as possible. When you check the tests they only check whether the `error` is there or not. We need to add actual logic verification. There should be few more PRs to make sure we are covered.

**Review by jakub-dzon — APPROVED** [review]

File: pkg/controller/virtualmachineimport/controller_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package virtualmachineimport
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestTemplates(t *testing.T) {
```
Comment (masayag) [inline pkg/controller/virtualmachineimport/controller_suite_test.go]: Perhaps TestController ?

Comment (pkliczewski) [reply pkg/controller/virtualmachineimport/controller_suite_test.go]: :+1:

**Review by machacekondra — APPROVED** [review]

**Pull request merged**